common/txmgr/broadcaster.go

```diff
@@ -558,21 +558,20 @@ func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) hand
 		eb.sequenceTracker.GenerateNextSequence(etx.FromAddress, *etx.Sequence)
 		return err, true
 	case client.Underpriced:
-		return eb.tryAgainBumpingGas(ctx, lgr, err, etx, attempt, initialBroadcastAt, retryCount+1)
+		return eb.tryAgainBumpingGas(ctx, lgr, err, errType, etx, attempt, initialBroadcastAt, retryCount+1)
 	case client.InsufficientFunds:
-		// NOTE: This bails out of the entire cycle and essentially "blocks" on
-		// any transaction that gets insufficient_funds. This is OK if a
-		// transaction with a large VALUE blocks because this always comes last
-		// in the processing list.
-		// If it blocks because of a transaction that is expensive due to large
-		// gas limit, we could have smaller transactions "above" it that could
-		// theoretically be sent, but will instead be blocked.
+		// NOTE:
+		// This can potentially happen during gas spike.
+		// We want to re-estimate the tx, and save the replaced attempt,
+		// and process it after the back off duration.
+		// This is done by tryAgainWithNewEstimation return retryable after saved the tx attempt,
+		// instead of calling handleInProgressTx() again
```

Comment:
`tryAgainWithNewEstimation`'s functionality can be read by the user, so we need to add information that isn't obvious to the reader. This sounds more accurate:
```suggestion
		// NOTE: This can happen either because of insufficient funds or during a gas spike
		// in combination with a high gas limit. In any case, we want to get a new estimation,
		// replace the existing attempt, and retry after the back off duration. The new attempt
		// needs to be stored immediately due to a DB constraint.
```